src/main/java/com/rackspace/salus/telemetry/ambassador/config/AmbassadorProperties.java

```diff
@@ -18,54 +18,75 @@
 
 import java.util.Collections;
 import java.util.List;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
 import lombok.Data;
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.stereotype.Component;
+import org.springframework.validation.annotation.Validated;
 
 @ConfigurationProperties("salus.ambassador")
 @Component
 @Data
+@Validated
 public class AmbassadorProperties {
+
+    @NotBlank
     String envoyRefreshInterval = "PT10S";
+
+    @NotNull
     long envoyRefreshParallelism = 5;
 
     /**
      * FOR DEVELOPMENT ONLY, disable TLS including mutual TLS authentication
      */
+    @NotNull
     boolean disableTls = false;
     // The following assume the working directory is $PROJECT_DIR$/dev
+    @NotBlank
     String certChainPath = "certs/out/ambassador.pem";
+    @NotBlank
     String trustCertPath = "certs/out/ca.pem";
+    @NotBlank
     String keyPath = "certs/out/ambassador-pkcs8-key.pem";
-
+    @NotBlank
     String vaultPkiRole = "telemetry-infra";
+    @NotBlank
     String externalName = "localhost";
+    @NotEmpty
     List<String> altExternalNames = Collections.singletonList("127.0.0.1");
 
+    @NotNull
```

Comment:
It's a primitive, so let's go with
```suggestion
    @Min(1)
```